tests/functional/test_endpoints.py

```diff
@@ -32,6 +34,32 @@
     'lex-runtime': 'runtime.lex',
 }
 
+BLACKLIST = [
+    'mobileanalytics',
+]
+
+
+def test_endpoint_matches_service():
+    backwards_renames = dict((v, k) for k, v in SERVICE_RENAMES.items())
+    session = get_session()
+    loader = session.get_component('data_loader')
+    expected_services = set(loader.list_available_services('service-2'))
+
+    pdir = os.path.dirname
+    endpoints_path = os.path.join(pdir(pdir(pdir(__file__))),
+                                  'botocore', 'data', 'endpoints.json')
+    with open(endpoints_path, 'r') as f:
+        data = json.loads(f.read())
+    for partition in data['partitions']:
+        for service in partition['services'].keys():
+            service = backwards_renames.get(service, service)
+            if service not in BLACKLIST:
+                yield _assert_endpoint_is_service, service, expected_services
+
+
+def _assert_endpoint_is_service(service, expected_services):
+    assert service in expected_services
```

Comment:
Maybe put a better message on failures instead of just saying it is not in expected services.